### DISCUSSION TITLE
Bluetooth/wifi synesthesia help.

### DISCUSSION THREAD

**Discussion by Sadvarsco** [onlaj/Piano-LED-Visualizer#455] [Q&A]
I got this thing working to a point. I even was able to correct some code here and there where there was some issues with the setup. But the one thing I cannot get working no matter how hard I try is the bluetooth portion of this.

I've seen videos of people using Synesthesia on their Ipad using this visualizer, and I'm not sure exactly what/where I'm going wrong. I believe I followed the setup on the given page, but still no long. Can anyone point me in the right direction?

I get as far as being able to see the Raspi via synesthesia app, but it cant connect.
The errors I see via raspi are

```
Jul 11 23:20:19 pi systemd-udevd[138]: hci0:64: Process '/usr/local/bin/connectall.rb' failed with exit code 1.
Jul 11 23:20:20 pi bluetoothd[511]: MIDI I/O: Failed to read initial request
```
SO I really just want to start from scratch with the bluetooth.

**Comment by onlaj** [comment]
Hey.
I put basically my whole knowledge about bluetooth in this setup [here](https://github.com/onlaj/Piano-LED-Visualizer/blob/master/Docs/btconnection.md)
The bluetooth connection is very problematic and I suggest investing into usb-usb midi device or just using RTP midi instead.

**Reply by Sadvarsco** [reply]
I did order a USB to USB. But I was hoping to get it running via bluetooth for other reasons. I'm going to try another fresh install and see if I can make any progress on either of those. 
Thanks.

**Comment by Sadvarsco** [comment]
> Hey. I put basically my whole knowledge about bluetooth in this setup [here](https://github.com/onlaj/Piano-LED-Visualizer/blob/master/Docs/btconnection.md) The bluetooth connection is very problematic and I suggest investing into usb-usb midi device or just using RTP midi instead.

I had a question. That guide (on that other page) is using rb, but the visualizer is using py. Are they just interchangeable? I can do all the stuff the author did in ruby in python? (including the connect all), or do I just need to copy the ruby code, which I assume means there will be both a connectall.py and a connectall.rb?

**Reply by onlaj** [reply]
> Are they just interchangeable?

Yes, they are. Some of it is also covered [here](https://github.com/onlaj/Piano-LED-Visualizer/blob/master/Docs/manual_installation.md)

Ruby version is just slower and requires additional package installed on OS.

```
% time python3 connectall.py

real    0m0.488s
user    0m0.383s
sys     0m0.102s

% time ruby connectall.rb

real    0m1.071s
user    0m0.843s
sys     0m0.227s
```
